lib/galaxy/webapps/galaxy/controllers/user.py

```diff
@@ -1177,33 +1178,33 @@ def reset_password( self, trans, email=None, **kwd ):
         message = None
         status = 'done'
         if kwd.get( 'reset_password_button', False ):
-            # Default to a non-userinfo-leaking response message
-            message = ( "Your reset request for %s has been received.  "
-                        "Please check your email account for more instructions.  "
-                        "If you do not receive an email shortly, please contact an administrator." % ( escape( email ) ) )
-            reset_user = trans.sa_session.query( trans.app.model.User ).filter( trans.app.model.User.table.c.email == email ).first()
-            if reset_user:
-                prt = trans.app.model.PasswordResetToken( reset_user )
-                trans.sa_session.add( prt )
-                trans.sa_session.flush()
-                host = trans.request.host.split( ':' )[ 0 ]
-                if host in [ 'localhost', '127.0.0.1', '0.0.0.0' ]:
-                    host = socket.getfqdn()
-                reset_url = url_for( controller='user',
-                                     action="change_password",
-                                     token=prt.token, qualified=True)
-                body = PASSWORD_RESET_TEMPLATE % ( host, reset_url, reset_url )
-                frm = trans.app.config.email_from
-                if frm is None:
-                    frm = 'galaxy-no-reply@' + host
-                subject = 'Galaxy Password Reset'
-                try:
-                    util.send_mail( frm, email, subject, body, trans.app.config )
-                    trans.sa_session.add( reset_user )
+            message = validate_email(trans, email, check_dup=False)
+            if not message:
+                # Default to a non-userinfo-leaking response message
+                message = ( "Your reset request for %s has been received.  "
+                            "Please check your email account for more instructions.  "
+                            "If you do not receive an email shortly, please contact an administrator." % ( escape( email ) ) )
+                reset_user = trans.sa_session.query( trans.app.model.User ).filter( func.lower(trans.app.model.User.table.c.email) == func.lower(email) ).first()
```

Comment:
Well, since we know they're intentionally unique *now*, how about we do a 2-pass check?  First check case-sensitive as it was before, and if there is no user, check insensitive?